controllers/runner_controller.go

```diff
@@ -398,6 +406,63 @@ func (r *RunnerReconciler) newPod(runner v1alpha1.Runner) (corev1.Pod, error) {
 		},
 	}
 
+	if len(runner.Spec.RunnerContainers) != 0 {
+		pod.Spec.Containers = runner.Spec.RunnerContainers
+	}
+
+	if len(runner.Spec.VolumeMounts) != 0 {
+		pod.Spec.Containers[0].VolumeMounts = append(pod.Spec.Containers[0].VolumeMounts, runner.Spec.VolumeMounts...)
+	}
+
+	if runner.Spec.RestartPolicy == "" {
+		pod.Spec.RestartPolicy = "OnFailure"
+	}
```

Comment:
Since the Controller is performing a restart upon completion of the runner's Pod, I don't think RestartPolicy should be overridable.

If RestartPolicy is set to "Always", the Controller will not be able to restart the Runner Pod properly.

https://github.com/summerwind/actions-runner-controller/blob/23538d43b310736ca5e7ebdfaeefa086ae7e7001/controllers/runner_controller.go#L183-L193

Reply:
got it